src/Symfony/Component/Cache/Traits/FilesystemCommonTrait.php

```
@@ -107,14 +108,20 @@ private function write(string $file, string $data, int $expiresAt = null)
             }
             fwrite($h, $data);
             fclose($h);
+            $unlink = true;
 
             if (null !== $expiresAt) {
                 touch($this->tmp, $expiresAt ?: time() + 31556952); // 1 year in seconds
             }
 
-            return rename($this->tmp, $file);
+            return !$unlink = !rename($this->tmp, $file);
         } finally {
             restore_error_handler();
+
+            if ($unlink) {
+                @unlink($this->tmp);
+                $this->tmp = null;
```

If we reset it to `null` here, is there any benefit at having it as an instance property rather than a local variable ?

---

the instance property was meant to avoid re-reading random bytes, but this was a benefit only if it was kept for next usage.

---

we reset to null only if rename fails

---

I agree, there is no need to reset this property here.

If unlink silently fails, it will anyway recompute a new value to write on a new file, thanks to this code:
https://github.com/symfony/symfony/blob/f14f23b1a4ff29a2de40c8572c392aa44d474834/src/Symfony/Component/Cache/Traits/FilesystemCommonTrait.php#L99-L108

---

which means this line is legit, since it allows bypassing some try/catch that we know for sure we'll hit

---

> which means this line is legit, since it allows bypassing some try/catch that we know for sure we'll hit

We will probably not hit it, as we just unlink the temporary file.

---

> > which means this line is legit, since it allows bypassing some try/catch that we know for sure we'll hit
> 
> We will probably not hit it, as we just unlink the temporary file.

I mean, if we reach tis point, it means that we were able to create a temporary file, write something in it, but moving it fails. It is highly probable that we will be able to link it, as moving failure is probably due to a restriction on the target path, not on the source path.